packages/trpc/server/routers/viewer/bookings/addGuests.handler.ts

```diff
@@ -1,52 +1,95 @@
 import { getUsersCredentialsIncludeServiceAccountKey } from "@calcom/app-store/delegationCredential";
+import { eventTypeMetaDataSchemaWithTypedApps } from "@calcom/app-store/zod-utils";
 import dayjs from "@calcom/dayjs";
-import { sendAddGuestsEmails } from "@calcom/emails";
+import { BookingEmailSmsHandler } from "@calcom/features/bookings/lib/BookingEmailSmsHandler";
 import EventManager from "@calcom/features/bookings/lib/EventManager";
+import { BookingRepository } from "@calcom/features/bookings/repositories/BookingRepository";
 import { PermissionCheckService } from "@calcom/features/pbac/services/permission-check.service";
 import { UserRepository } from "@calcom/features/users/repositories/UserRepository";
 import { extractBaseEmail } from "@calcom/lib/extract-base-email";
 import { parseRecurringEvent } from "@calcom/lib/isRecurringEvent";
+import logger from "@calcom/lib/logger";
 import { getTranslation } from "@calcom/lib/server/i18n";
 import { prisma } from "@calcom/prisma";
 import { MembershipRole } from "@calcom/prisma/enums";
 import type { BookingResponses } from "@calcom/prisma/zod-utils";
+import { eventTypeBookingFields } from "@calcom/prisma/zod-utils";
 import type { CalendarEvent } from "@calcom/types/Calendar";
 
 import { TRPCError } from "@trpc/server";
 
 import type { TrpcSessionUser } from "../../../types";
 import type { TAddGuestsInputSchema } from "./addGuests.schema";
 
+type TUser = Pick<NonNullable<TrpcSessionUser>, "id" | "email" | "organizationId"> &
+  Partial<Pick<NonNullable<TrpcSessionUser>, "profile">>;
+
 type AddGuestsOptions = {
   ctx: {
-    user: NonNullable<TrpcSessionUser>;
+    user: TUser;
   };
   input: TAddGuestsInputSchema;
+  emailsEnabled?: boolean;
 };
-export const addGuestsHandler = async ({ ctx, input }: AddGuestsOptions) => {
+
+type Booking = NonNullable<Awaited<ReturnType<BookingRepository["findByIdIncludeDestinationCalendar"]>>>;
+type OrganizerData = Awaited<ReturnType<typeof getOrganizerData>>;
+
+export const addGuestsHandler = async ({ ctx, input, emailsEnabled = true }: AddGuestsOptions) => {
```

Comment:
We call it `addGuestsHandler` but we are adding attendees now and also in schema `packages/trpc/server/routers/viewer/bookings/addGuests.schema.ts` we mix attendees with guests so bit confusing.